forge-ai/src/main/java/forge/ai/AiAttackController.java

```diff
@@ -511,6 +508,85 @@ public boolean apply(final Card c) {
+        List<String> bandsWithString = Arrays.asList("Bands with Other Legendary Creatures",
+                "Bands with Other Creatures named Wolves of the Hunt",
+                "Bands with Other Dinosaurs");
+
+        List<Card> bandingCreatures = new CardCollection();
+
+        if (test == null) {
+            bandingCreatures.addAll(notNeededAsBlockers(player.getCreaturesInPlay()));
+            bandingCreatures = CardLists.filter(bandingCreatures, card -> card.hasKeyword(Keyword.BANDING) || card.hasAnyKeyword(bandsWithString));
+
+            // filter out anything that can't legally attack or is already declared as an attacker
+            bandingCreatures = CardLists.filter(bandingCreatures, card -> !combat.isAttacking(card) && CombatUtil.canAttack(card));
+        } else {
+            // Test a specific creature for Banding
+            if (test.hasKeyword(Keyword.BANDING) || test.hasAnyKeyword(bandsWithString)) {
+                bandingCreatures.add(test);
+            }
+        }
+
+        // respect global attack constraints
+        GlobalAttackRestrictions restrict = GlobalAttackRestrictions.getGlobalRestrictions(ai, combat.getDefenders());
+        int attackMax = restrict.getMax();
+        if (attackMax >= combat.getAttackers().size()) {
+            return;
+        }
+
+        if (!bandingCreatures.isEmpty()) {
+            List<String> evasionKeywords = Arrays.asList("Flying", "Horsemanship", "Shadow", "Plainswalk", "Islandwalk",
+                    "Forestwalk", "Mountainwalk", "Swampwalk");
+
+            // TODO: Assign to band with the best attacker for now, but needs better logic.
+            CardCollection attackers = combat.getAttackers();
+            for (Card c : bandingCreatures) {
+                Card bestBand;
+
+                if (c.getNetPower() <= 0) {
+                    // Don't band a zero power creature if there's already a banding creature in a band
+                    attackers = CardLists.filter(attackers, card -> combat.getBandOfAttacker(card).getAttackers().size() == 1);
+                }
+
+                Card bestAttacker = ComputerUtilCard.getBestCreatureAI(attackers);
+                if (c.hasKeyword("Bands with Other Legendary Creatures")) {
+                    bestBand = ComputerUtilCard.getBestCreatureAI(CardLists.getType(attackers, "Legendary"));
+                } else if (c.hasKeyword("Bands with Other Dinosaurs")) {
+                    bestBand = ComputerUtilCard.getBestCreatureAI(CardLists.getType(attackers, "Dinosaur"));
+                } else if (c.hasKeyword("Bands with Other Creatures named Wolves of the Hunt")) {
+                    bestBand = ComputerUtilCard.getBestCreatureAI(CardLists.filter(attackers, CardPredicates.nameEquals("Wolves of the Hunt")));
+                } else if (bestAttacker != null && !c.hasAnyKeyword(evasionKeywords) && bestAttacker.hasAnyKeyword(evasionKeywords)) {
```

Comment:
When would bestAttacker be null - only if there were no attackers at all right?

So this should just be a guard clause at the top to avoid tons of calls 🤔

Reply:
Yep, removed the null check since you added a guard above :+1: